pkg/kubelet/dockershim/libdocker/kube_docker_client.go

```diff
@@ -595,13 +617,20 @@ func (d *kubeDockerClient) holdHijackedConnection(tty bool, inputStream io.Reade
 		close(stdinDone)
 	}()
 
+	timeoutPassed := make(chan struct{})
+	if timeout > 0 {
+		time.AfterFunc(timeout, func() { close(timeoutPassed) })
```

Comment:
It's probably a good idea to store the `time.Timer` returned by `time.AfterFunc` so we can `defer timeoutTimer.Stop()` - this will ensure we don't have timers with long timeouts sticking around for a while after the exec has returned successfully. WDYT?

Reply:
Yes, stopping the timer on function exit is really a good idea.